File: tools/conf/output/go/PartTypeConfig.go
package conf

import (
	//"bufio"
	"encoding/json"
	"fmt"
	//"io"
	"io/ioutil"
	"os"
	//"strings"
)

type PartTypeConfigElem struct {
	Id int
	Parent int
	Name string
	Icon string
	IconDown string
	Gender int
	
}

type PartTypeConfig0 struct {
	Dic map[int]*PartTypeConfigElem
	Arr []*PartTypeConfigElem
}

var (
	PartTypeConfig PartTypeConfig0
)

func loadPartTypeConfig(path string) {
	file, err := os.Open(path)
	defer func() {
		if file != nil {
			file.Close()
		}
	}()
	if err != nil {
		panic(fmt.Sprint("load config error, path=", path))
	}

	PartTypeConfig.Dic = make(map[int]*PartTypeConfigElem)
	PartTypeConfig.Arr = make([]*PartTypeConfigElem, 0, 32)

	bt,err  :=  ioutil.ReadAll(file)
	json.Unmarshal(bt, &PartTypeConfig.Arr)
	for _,conf := range PartTypeConfig.Arr{
		PartTypeConfig.Dic[conf.Id] = conf
	}
	
}

File: client/src/game/conf/PartTypeConfig.ts
class PartTypeConfig {
    private static _dic:Dictionary<PartTypeConfig> = new Dictionary<PartTypeConfig>();
    private static _arr:Array<PartTypeConfig> = new Array<PartTypeConfig>();

	 public Id :number ;
     public Parent :number ;
     public Name :string ;
     public Icon :string ;
     public IconDown :string ;
    
    

    
    public constructor(){

    }


    public static get Dic():Dictionary<PartTypeConfig>{
        return  PartTypeConfig._dic;
    }
    public static get Arr():Array<PartTypeConfig>{
        return PartTypeConfig._arr;
    }
    public static parse(json:any[]){
        if(json && json.length > 0){
			for(var i = 0; i < json.length; i++){
				let config:PartTypeConfig = <PartTypeConfig>json[i];
				PartTypeConfig._arr.push(config);
				PartTypeConfig._dic.add(config.Id,config);
			}
        }else{
            console.error("PartTypeConfig loader fail!");
        }
    }
}